infoschema/tables_test.go

```diff
@@ -449,7 +449,17 @@ func (s *testTableSuite) TestSomeTables(c *C) {
 		testkit.Rows("def mysql tbl def mysql stats_meta table_id 1 <nil> <nil> <nil> <nil>"))
 	tk.MustQuery("select * from information_schema.STATISTICS where TABLE_NAME='columns_priv' and COLUMN_NAME='Host';").Check(
 		testkit.Rows("def mysql columns_priv 0 mysql PRIMARY 1 Host A <nil> <nil> <nil>  BTREE  "))
-	tk.MustQuery("select * from information_schema.USER_PRIVILEGES where PRIVILEGE_TYPE='Select';").Check(testkit.Rows("'root'@'%' def Select YES"))
+
+	//test the privilege of new user for information_schema
+	tk.MustExec("create user statistics_tester")
+	tk.MustExec("flush privileges")
+	tk1 := testkit.NewTestKit(c, s.store)
+	tk1.MustExec("use information_schema")
+	c.Assert(tk1.Se.Auth(&auth.UserIdentity{
+		Username: "statistics_tester",
+		Hostname: "127.0.0.1",
+	}, nil, nil), IsTrue)
+	tk1.MustQuery("select * from information_schema.STATISTICS;").Check([][]interface{}{})
```

Comment:
Suggestion:
Adding one more test case, grant necessary roles for new user, and it should get non-empty result.

Reply:
PTAL @imtbkcat

Reply:
@reafans try this case:
```
create role r_schema;
grant all privileges on *.* to r_schema;
grant r_schema to statistics_tester;
```
then create a session as `statistics_tester` and run query below.

```
set role r_schema;
select * from information_schema.STATISTICS;
```
it should get non-empty result.

Reply:
done，PTAL